Standardizing tags

inquisitev opened the pull request:
https://github.com/goatshriek/stumpless/issues/372
- [x] find and replace of the variants
- [x] static analysis script

<!--
Make sure that your pull request follows the guidelines specified in the
[Guidelines for Contributing](../docs/CONTRIBUTING.md). 
-->

inquisitev committed:
Add GH action to enforce @since release vX.X.X, replace all offending

replace all tags

add action

add comments to action

typo on return var

dont local

exit 1 with no ret

fix since syntax

use bashxargs

should fail test

exit zero when passing, 1 when failing

clean last tags

changelog

goatshriek commented on .github/workflows/analysis.yml:
```diff
@@ -70,3 +70,27 @@ jobs:
+      - name: "@since standardization"
+        run: |
+          # look for all instances of @since, then look for any that do not meet the required pattern. @since release x.x.x
+          if grep --include=\*.{c,h} -rnw . -e ".*@since .*" | grep -qv ".* @since release v[[:digit:]]\+\.[[:digit:]]\+\.[[:digit:]].*"; then
```
add cpp and hpp to the file extensions searched, so that the test helpers and the like can also be standardized moving forward

goatshriek commented on .github/workflows/analysis.yml:
```diff
@@ -70,3 +70,27 @@ jobs:
+            # at this point the check has failed, but we want to give some info to the user. i have found it difficult to get the status code
+            # and print out the offending lines without this duplication
+            # get all instances of @since, offending and not offending
+            grep --include=\*.{c,h} -rnw . -e "@since .*" | \
```
cpp and hpp here as well, of course

goatshriek commented on ChangeLog.md:
```diff
@@ -8,7 +8,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 fixes, check out the
 [roadmap](https://github.com/goatshriek/stumpless/blob/master/docs/roadmap.md).
-
+### Added 
+ - @since format check enforcement in CI pipeline.
```
This should go under the 2.2.0 heading, instead of above it

inquisitev replied:
ohh i saw the unreleased tag and assumed thats where it should go. should i remove that section since there is no next/staging branch?

should the date be updated aswell on the 2.2.0 tag?

goatshriek replied:
You're right, the "Unreleased" heading is a little confusing - it's a holdover from an earlier changelog that also included the roadmap content. Feel free to remove the heading (line 7) itself, just make sure you leave the paragraph and link to the roadmap in.

Ideally please update the 2.2.0 tag date as well, though this is only critical when a release is cut, which will likely be sometime in November for 2.2.0.

Don't worry too much about trying to do force pushes or amends - the pull will be squashed when it's merged.

inquisitev replied:
done, moved the entry and removed unrelease tag.

goatshriek reviewed (changes requested):
Thanks very much for putting this together! Only a few minor tweaks needed, and then this will be ready to go.

inquisitev committed:
Add GH action to enforce @since release vX.X.X, replace all offending

replace all tags

add action

add comments to action

typo on return var

dont local

exit 1 with no ret

fix since syntax

use bashxargs

should fail test

exit zero when passing, 1 when failing

clean last tags

changelog

inquisitev commented:
dang i messed up that amend...https://github.com/goatshriek/stumpless/pull/373/commits/eb68750fdb9c43afc776412ad9ed2497c21fdfbe

inquisitev committed:
changelog, move entry and remove unreleased
